124-stretch-cluster.md

```diff
@@ -0,0 +1,1795 @@
+
+The central operator has write access to remote clusters via kubeconfig and creates resources directly.
+Remote operators are responsible only for reconciling StrimziPodSet resources (creating/updating pods).
+
+### Configuration Design Rationale
+
+#### Why Stretch Configuration is at Operator Level
+
+The stretch cluster configuration (`STRIMZI_CENTRAL_CLUSTER_ID`, `STRIMZI_REMOTE_KUBE_CONFIG`, plugin settings) is specified at the cluster operator level rather than at the Kafka CR or KafkaNodePool level.
+This design decision addresses several important architectural and operational concerns.
+
+##### Infrastructure-Level Concern
+
+Stretch cluster functionality requires infrastructure-level connectivity and authentication to remote Kubernetes clusters.
+The operator needs kubeconfig credentials and API server endpoints to create resources in remote clusters.
+These credentials grant cluster wide access to Kubernetes APIs and are managed by platform administrators, not application developers.
+
+Placing these credentials at the operator level:
+
+- Aligns with Kubernetes security model where operators run with ServiceAccounts that have cluster-level permissions
+- Avoids embedding sensitive kubeconfig credentials in application level CRs (Kafka, KafkaNodePool).
+- Simplifies credential rotation by updating operator deployment once rather than every Kafka CR
+- Prevents credential sprawl across multiple Kafka instances
+
+##### Single Networking Provider Per Operator
+
+The networking plugin (NodePort, LoadBalancer, MCS) is loaded once at operator startup via Java ServiceLoader.
+The plugin JAR is mounted as a volume in the operator deployment.
+The plugin's initialization code runs once when the operator starts.
+
+This means all Kafka clusters managed by a single operator instance must use the same networking provider.
+Different networking providers cannot coexist in a single operator instance.
+If networking provider were configurable per Kafka CR, the operator would need to:
+- Load and initialize multiple plugin JARs dynamically
+- Maintain separate ClassLoaders for each plugin to avoid conflicts
+- Handle plugin initialization failures on a per-cluster basis
+- Manage plugin lifecycle (loading/unloading) during reconciliation
+
+This adds significant complexity and creates potential ClassLoader isolation issues.
+
+
+**Example scenario:** If networking provider were per Kafka CR, the operator would need to:
+
+Load and initialize multiple plugin JARs dynamically.
+Maintain separate ClassLoaders for each plugin to avoid conflicts.
+Handle plugin initialization failures on a per cluster basis.
+Manage plugin lifecycle (loading/unloading) during reconciliation.
+This adds significant complexity and creates potential ClassLoader isolation issues.
+
+##### Consistent Cluster Topology
+
+The `STRIMZI_CENTRAL_CLUSTER_ID` and `STRIMZI_REMOTE_KUBE_CONFIG` define the operator's view of the multi cluster topology.
```

Comment:
This likely needs some proper structure. You should not have a single kubeconfig configuration. The purpose of the central cluster ID is not clear either.